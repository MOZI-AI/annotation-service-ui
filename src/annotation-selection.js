import React from 'react';
import { Form, Radio, Checkbox } from 'antd';


export class AnnotationSelection extends React.Component {

    render() {
        return (
            <React.Fragment>
                <h4 style={{ marginBottom: '15px' }}>Annotations: </h4>
                {
                    this.props.availableAnnotations.map((annotation, index) =>
                        <div key={annotation}>
                            <Checkbox name={annotation} onChange={e => this.props.onAnnotationsChanged(e)}> {annotation} </Checkbox>
                            {this.props.selectedAnnotations.some(a => a.name === annotation) &&
                                <Form style={{ paddingLeft: '25px', marginBottom: '30px', marginLeft: '5px', paddingTop: '15px', borderLeft: 'dashed 1px #ddd' }} layout='inline'>
                                    <h3>{annotation} Filters</h3>
                                    <Radio.Group name='option' onChange={e => this.props.onAnnotationFilterChanged(annotation, e)}>
                                        <Radio value={1}> Option One </Radio>
                                        <Radio value={2}> Option Two </Radio>
                                        <Radio value={3}> Option Three </Radio>
                                    </Radio.Group>
                                </Form>}
                        </div>
                    )
                }

            </React.Fragment>
        )
    }

}