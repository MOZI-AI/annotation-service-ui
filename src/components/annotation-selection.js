import React from "react";

import { Checkbox, FormControlLabel } from "@material-ui/core";

const filterFormStyle = {
  paddingLeft: "25px",
  marginLeft: "5px",
  borderLeft: "dashed 1px #ddd"
};

export class AnnotationSelection extends React.Component {
  isAnnotationSelected = annotation =>
    this.props.selectedAnnotations.some(sa => sa.name === annotation);

  render() {
    return (
      <React.Fragment>
        <h4>Annotations: </h4>

        {this.props.availableAnnotations.map(a => (
          <React.Fragment key={a.key}>
            <FormControlLabel
              value="0"
              control={
                <Checkbox
                  name={a.key}
                  onChange={e =>
                    this.props.handleAnnotationsChanged(
                      e.target.checked,
                      e.target.name
                    )
                  }
                />
              }
              label={a.name}
            />
            <br />
            {this.isAnnotationSelected(a.key) && (
              <div style={filterFormStyle} layout="inline">
                {a.fitlerForm(a.defaults, filter =>
                  this.props.handleFilterChanged(a.key, filter)
                )}
              </div>
            )}
          </React.Fragment>
        ))}
      </React.Fragment>
    );
  }
}
