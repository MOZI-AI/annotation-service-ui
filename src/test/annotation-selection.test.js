import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { AnnotationSelection } from '../annotation-selection'
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


describe('<AnnotationSelection />', () => {

    it('renders correctly', () => {
        const tree = renderer.create(<AnnotationSelection availableAnnotations={['a', 'b']} selectedAnnotations={[]} />).toJSON()
        expect(tree).toMatchSnapshot()
    });


    it('renders a checkbox for each selected annotation', () => {
        const wrapper = shallow(<AnnotationSelection availableAnnotations={['a', 'b']} selectedAnnotations={[]} />)
        expect(wrapper.find('Checkbox').length).toEqual(2)
    })


    it('renders a filter form for each selected annotation', () => {
        const wrapper = shallow(<AnnotationSelection availableAnnotations={['a', 'b']} selectedAnnotations={[{ name: 'a', filter: {} }]} />)
        expect(wrapper.find('Checkbox + Form').length).toEqual(1)
    })


})