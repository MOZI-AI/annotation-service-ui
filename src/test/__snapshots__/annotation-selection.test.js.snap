// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnnotationSelection /> renders correctly 1`] = `
Array [
  <h4
    style={
      Object {
        "marginBottom": "15px",
      }
    }
  >
    Annotations: 
  </h4>,
  <div>
    <label
      className="ant-checkbox-wrapper"
    >
      <span
        className="ant-checkbox"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-checkbox-input"
          name="a"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
         
        a
         
      </span>
    </label>
  </div>,
  <div>
    <label
      className="ant-checkbox-wrapper"
    >
      <span
        className="ant-checkbox"
        style={Object {}}
      >
        <input
          checked={false}
          className="ant-checkbox-input"
          name="b"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          type="checkbox"
        />
        <span
          className="ant-checkbox-inner"
        />
      </span>
      <span>
         
        b
         
      </span>
    </label>
  </div>,
]
`;
